File: p8.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.IO;

namespace ConsoleApp1
{
    class p8
    {
        public static long product(int a, List<int> nums)
        {
            long total = 1;
            for (int i = a; i < a+13; i++)
            {
                total *= nums[i];
            }
            return total;
        }
        public static void Main(String[] args)
        {
            List<int> nums = new List<int>();
            using (var reader = new StreamReader("8.csv"))
            {
                
                while (!reader.EndOfStream)
                {
                    var line = reader.ReadLine();
                    var values = line.Split(',');
                    int[] n = Array.ConvertAll(values, s => int.Parse(s));
                    foreach (int a in n)
                    {
                        nums.Add(a);
                    }
                }
            }
            long highest = 0;
            for (int i = 0; i < 985; i++)
            {
                long currentProduct = product(i, nums);
                if (currentProduct > highest) { highest = currentProduct; }
            }
            Console.WriteLine(highest);
        }
    }
}

File: p8.py
import csv
matrix = []
with open('8.csv', newline='') as csvfile:
    itemReader = csv.reader(csvfile, delimiter=',')
    for row in itemReader:
        for item in row:
            matrix.append(int(item))

def product(a):
    total = 1
    for i in range(a, a+13):
        total *= matrix[i]
    return total

highest = 0
for i in range(0, 985):
    currentProduct = product(i)
    if currentProduct > highest:
        highest = currentProduct

print(highest)
